### DISCUSSION TITLE
C++ Use the typedef name for anonymous struct / union / enum, instead of the auto-generated name

### DISCUSSION THREAD

**Discussion by terminatorul** [universal-ctags/ctags#4021] [Ideas]
Hello

Consider an anonymous structure in a typdef declaration:
```C
typedef struct
{
    char name[256];
    long count;
    void *data;
}
    PluginList;
```

ctags generates an anonymous name like `__anon9f26d2460108`  for the structure, and also references it for the members:

![image](https://github.com/universal-ctags/ctags/assets/378924/cb7dfba5-831a-4b82-a04e-487ed9fa5a70)

Can ctags use the typedef name `PluginList` as the structure name for this case ?

In C++ this is already specified as a special case by the language, where the typedef is used to denote the class or enum type for linkage purposes (and the name receives the same linkage as the class or namespace of which it is a member).

This means objects declared "extern" (or static), of this struct type, receive the class type PluginList directly.

However I think ctags should do the same thing C as well.

**Comment by masatake** [comment]
Ctags cannot do it.

Like #4015, this should be done in post-process.
Ctags should provide raw data AT LEAST. This is one of the mottos of Universal Ctags.
If ctags reports PluginList as a struct, client tools cannot know PluginList is a typedef. However, it is o.k. that ctags reports PluginList as a struct in addition to reporting PluginList as a typedef. Exuberant Ctags provides a feature named 'extra tags'. Universal Ctags, an unofficial fork, inherit the concept.

Let's assume we have already implemented such a feature in a ctags implementation named "Ideal Ctags".

The feature isn't enabled by default. To turn it on, specify `--extras-C++=+{typedefContaction}.` 

`./ctags --extras-C++=+{typedefContracton} -o - ` emits:


```
__anon88eebcb90108	foo.cxx	/^{$/;"	struct	file:	extras:fileScope,anonymous
name	foo.cxx	/^    char name[256];$/;"	member	scope:struct:__anon88eebcb90108	typeref:typename:char[256]	file:	access:public	extras:fileScope	nth:0
count	foo.cxx	/^    long count;$/;"	member	scope:struct:__anon88eebcb90108	typeref:typename:long	file:	access:public	extras:fileScope	nth:1
data	foo.cxx	/^    void *data;$/;"	member	scope:struct:__anon88eebcb90108	typeref:typename:void *	file:	access:public	extras:fileScope	nth:2
PluginList	foo.cxx	/^    PluginList;$/;"	typedef	typeref:struct:__anon88eebcb90108	file:	extras:fileScope
PluginList	foo.cxx	/^    PluginList;$/;"	struct	file:	extras:fileScope,typedefContraction
foo.cxx	foo.cxx	1;"	file	extras:inputFile	epoch:1718376238
```

Implementing this one may be easy.
However, you may want to do the same on the scope fields of the members like:
```
count	foo.cxx	/^    long count;$/;"	member	scope:struct:PluginList	typeref:typename:long	file:	access:public	extras:fileScope	nth:1
```

It is harder to implement. However, it may be possible.
